plugins/videobasedtracker/UVCImageSource.cpp

```diff
@@ -245,20 +245,20 @@ namespace vbtracker {
         // upon request in the retrieveColor() method.
 
         // Copy the frame, then look for the lock
-        uvc_frame_t *copied_frame = uvc_allocate_frame(frame->data_bytes);
-        uvc_duplicate_frame(frame, copied_frame);
+        Frame_ptr copied_frame(uvc_allocate_frame(frame->data_bytes),
+                               &uvc_free_frame);
+        uvc_duplicate_frame(frame, copied_frame.get());
 
         std::lock_guard<std::mutex> lock(mutex_);
-        frames_.push(copied_frame);
+        frames_.emplace(copied_frame.release(), &uvc_free_frame);
         if (frames_.size() > 100) {
             std::cerr << "WARNING! Dropping frames from video tracker as they "
                          "are not being processed fast enough! This will "
-                         "disrupt tracking.";
-            while (!frames_.empty()) {
-                uvc_free_frame(frames_.front());
-                frames_.pop();
-            }
+                         "disrupt tracking."
+                      << std::endl;
+            frames_ = std::queue<Frame_ptr>(); //< clear the queue
```

Comment:
@toastedcrumpets You could use `std::vector` instead. It has `push_back()` and `pop_back()` methods as well as `clear()`.